Core/Plugins/OpenHardwareMonitor/Hardware/Ring0.cs

```diff
@@ -117,10 +118,29 @@ private static bool ExtractDriver(string fileName)
 
         public static void Open()
         {
-            // no implementation for unix systems
-            int p = (int)Environment.OSVersion.Platform;
-            if ((p == 4) || (p == 128))
-                return;
+			System.PlatformID p = Environment.OSVersion.Platform;
+
+            // try loading shipped kernel drivers on Unix
+			// (will probably only work on Linux through)
+			if (p == PlatformID.Unix) {
+				// try loading the `msr` kernel module on Linux (required on ost kernels)
+				Process modprobe = new Process();
+				try {
+					modprobe.StartInfo.FileName  = "modprobe";
+					modprobe.StartInfo.Arguments = "msr";
+					modprobe.Start();
+					modprobe.WaitForExit();

```

Comment:
No, I left this unchecked intentionally. While most systems will require this driver to be loaded at run-time, some may already have it compiled into the kernel. In that case loading the driver will fail, but the MSR interface will be available nevertheless.

I'll add a comment about this instead…